File: backend/Api.EscolaIdiomas/Api.EscolaIdiomas.Domain/Models/Alunos/Aluno.cs
namespace Api.EscolaIdiomas.Domain.Models.Alunos
{
    public class Aluno
    {
        public long Id { get; set; }

        public string Nome { get; set; } = string.Empty;

        public string? Sobrenome { get; set; }

        public DateTime? DataDeNascimento { get; set; }

        public string? Email { get; set; }

        public string? Telefone { get; set; }

        public DateTime? DataMatricula { get; set; }

        public bool Ativo { get; set; }
    }
}

File: frontend/src/types/Aluno.ts
export type Aluno = {
  id: number;
  nome: string;
  sobrenome?: string;
  email?: string;
  telefone?: string;
  dataDeNascimento?: Date;
  dataMatricula?: Date;
  ativo?: boolean;
};

// Tipo para a resposta da API que agora retorna todos os campos
export type AlunoResumo = {
  id: number;
  nome: string;
  sobrenome?: string;
  email?: string;
  telefone?: string;
  dataDeNascimento?: string; // Vem como string da API
  dataMatricula?: string; // Vem como string da API
  ativo: boolean;
};